### PR TITLE
Debounce debounce

### PR DESCRIPTION
**Pull request by nikku**
Some internal cleanups:

* Drop use of services within components but rely on hooks instead
* Prepare for cancelable debounced functions, to be added one day, maybe (https://github.com/bpmn-io/min-dash/pull/20)

### CODE REVIEW

File: packages/form-js-editor/src/render/hooks/useDebounce.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import useService from './useService';
+
+
+export default function useDebounce(fn, dependencies=[]) {
+
+  const debounce = useService('debounce');
+
+  const callback = useMemo(() => {
+    return debounce(fn);
+  }, [...dependencies, fn ]);
```
Comment (barmac) [inline packages/form-js-editor/src/render/hooks/useDebounce.js:15]:
```suggestion
  }, dependencies);
```

Comment (barmac) [reply packages/form-js-editor/src/render/hooks/useDebounce.js:15]: Otherwise it's flushed on each render if `fn` is defined within the render function.

**Review by barmac — APPROVED** [review]
I love it.

**Comment by nikku** [comment]
@philippfromme Please have a look.

**Pull request merged**